src/routers/alpha-router/quoters/base-quoter.ts

```diff
@@ -200,14 +202,16 @@ export abstract class BaseQuoter<
     const tokens = _.flatMap(pools, (pool) => [pool.token0, pool.token1]);
 
     const tokenValidationResults =
-      await this.tokenValidatorProvider.validateTokens(tokens);
+      await this.tokenValidatorProvider.validateTokens(
+        tokens.map((token) => token.wrapped)
```

Comment:
curious why is this change needed here? will it affect V2 Pair?

Reply:
Aw, this is because in V4, we will support native currency routing/quoting. In V2 and V3, all uniswap pools can only support ERC20 tokens, but not native currencies. In V4, any Uniswap pool can support native currency. So that means, data structure wise, tokens will change its data type from `Token` to `Currency`, where `Currency = Token | NativeCurrency`. This means, since V2 and V3 don't support native currency pool, so far we can just safely wrap the token with type `Currency` to type `Token`. But in v4, we will need to dedicate our effort in supporting native currency routing in separate PRs, since I expect that to be non-trivial amount of work.